editor/plugins/version_control_editor_plugin.cpp

```diff
@@ -501,55 +1241,256 @@ VersionControlEditorPlugin::VersionControlEditorPlugin() {
 	commit_message->set_wrap_enabled(true);
 	commit_message->connect("text_changed", this, "_update_commit_button");
 	commit_message->connect("gui_input", this, "_commit_message_gui_input");
-	commit_box_vbc->add_child(commit_message);
+	commit_area->add_child(commit_message);
+
 	ED_SHORTCUT("version_control/commit", TTR("Commit"), KEY_MASK_CMD | KEY_ENTER);
 
 	commit_button = memnew(Button);
 	commit_button->set_text(TTR("Commit Changes"));
 	commit_button->set_disabled(true);
-	commit_button->connect("pressed", this, "_send_commit_msg");
-	commit_box_vbc->add_child(commit_button);
-
-	commit_status = memnew(Label);
-	commit_status->set_align(Label::ALIGN_CENTER);
-	commit_box_vbc->add_child(commit_status);
-
-	version_control_dock = memnew(PanelContainer);
+	commit_button->connect("pressed", this, "_commit");
+	commit_area->add_child(commit_button);
+
+	version_commit_dock->add_child(memnew(HSeparator));
+
+	HBoxContainer *commit_list_hbc = memnew(HBoxContainer);
+	version_commit_dock->add_child(commit_list_hbc);
+
+	Label *commit_list_label = memnew(Label);
+	commit_list_label->set_text(TTR("Commit List"));
+	commit_list_label->set_h_size_flags(Control::SIZE_EXPAND_FILL);
+	commit_list_hbc->add_child(commit_list_label);
+
+	commit_list_size_button = memnew(OptionButton);
+	commit_list_size_button->set_tooltip(TTR("Commit List Size"));
```

Comment:
Use "Sentence case" for tooltips:
```suggestion
	commit_list_size_button->set_tooltip(TTR("Commit list size"));
```